### PR TITLE
Better implementation of the Musician trait

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Any talented Musician can now use any instrument to lift
spirits, and ease burdens. They can now also use the space piano and minimog
to grant people the Good Music buffs, just like a handheld instrument.
/:cl:

A living mob with TRAIT_MUSICIAN will now apply the Good Music
status effect to everyone who can hear the music. The buff will
now apply even to players who have disabled instrument sounds.

The Good Music status effect lasts for 6 seconds after the
musician stops playing.

Added STATUS_EFFECT_REFRESH that just refreshes the duration
of a status effect, rather than instancing a new one.

### CODE REVIEW

**Comment by Mickyan** [comment]
This changes intended functionality so I wouldn't just call it a code improvement

I don't mind too much either way, just saying

**Comment by actioninja** [comment]
I agree, this is more than a code improvement. It is a more logical and better implementation though.

**Comment by Farquaar** [comment]
Seems like a good change to me. The musical tuner just took up space in the inventory anyway.

File: code/datums/status_effects/buffs.dm [hunk 1]
```diff
@@ -541,3 +541,16 @@
 			else if(isanimal(L))
 				var/mob/living/simple_animal/SM = L
 				SM.adjustHealth(-3.5, forced = TRUE)
+
+/datum/status_effect/good_music
+	id = "Good Music"
+	alert_type = null
+	duration = 6 SECONDS
+	tick_interval = 1 SECONDS
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/status_effects/buffs.dm]: Tick interval is already 1 second

Comment (coiax) [reply code/datums/status_effects/buffs.dm]: Is it that bad to be explicit?

Comment (ExcessiveUseOfCobblestone) [reply code/datums/status_effects/buffs.dm]: should be fine in this instance.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-2). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**